wallet/dapps-communication.md

```diff
@@ -1,11 +1,366 @@
+
+#### Account
+
+##### List
+
+This event is used by the webpage to list known accounts by the extension.
+
+| Direction            | Type       | Format                                      | Example                                                         |
+| -------------------- | ---------- | -------------------------------------------- | --------------------------------------------------------------- |
+| Webpage to extension | `account.list` | None  | <pre><code class="highlight-source-json">null</code></pre> |
+| Extension to webpage | `account.list.response` | <pre><code class="highlight-source-yaml">type:  array<br>items:<br>  type: object<br>  properties:<br>    address:<br>      type: string<br>    name:<br>      type: string<br>  required:<br>    - address</code></pre> | <pre><code class="highlight-source-json">[<br>  {<br>    "address": "A12...",<br>    "name": "Account 1"<br>  },<br>  {<br>    "id": "A12..."<br>  }<br>]</code></pre> |
+
+##### Balance
+
+| Direction            | Type       | Format                                      | Example                                                         |
+| -------------------- | ---------- | -------------------------------------------- | --------------------------------------------------------------- |
+| Webpage to extension | `account.balance` | <pre><code class="highlight-source-yaml">type: object<br>properties:<br>  address:<br>  type: string<br>required:<br>  - address</code></pre>  | <pre><code class="highlight-source-json">{<br>  "address": "A12..."<br>}</code></pre> |
+| Extension to webpage | `account.balance.response` | <pre><code class="highlight-source-yaml">type: object<br>properties:<br>  balance:<br>    type: string<br>    format: bigInt<br>required:<br>  - balance</code></pre> | <pre><code class="highlight-source-json">{<br>  "balance": 1000n<br>}</code></pre> |
+
+##### Delete
+
+| Direction            | Type       | Format                                      | Example                                                         |
+| -------------------- | ---------- | -------------------------------------------- | --------------------------------------------------------------- |
+| Webpage to extension | `account.delete` | <pre><code class="highlight-source-yaml">type: object<br>properties:<br>  address:<br>    type: string<br>required:<br>  - address</code></pre>  | <pre><code class="highlight-source-json">{<br>  "address": "A12..."<br>}</code></pre> |
+| Extension to webpage | `account.delete.response` | <pre><code class="highlight-source-yaml">type: object<br>properties:<br>  response:<br>    type: string<br>    enum: ["OK", "REFUSED", "ERROR"]<br>  error:<br>    type: string<br>required:<br>  - response</code></pre> | <pre><code class="highlight-source-json">{<br>  "response": "REFUSED"<br>}</code></pre> |
```

Comment:
error property is missing

Reply:
This property is optional (not required).